luskan/src/main/java/org/opencadc/luskan/SiteLocationsConverter.java

```diff
@@ -0,0 +1,118 @@
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  : 5 $
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.luskan;
+
+import ca.nrc.cadc.tap.parser.navigator.ExpressionNavigator;
+import ca.nrc.cadc.tap.parser.navigator.FromItemNavigator;
+import ca.nrc.cadc.tap.parser.navigator.ReferenceNavigator;
+import ca.nrc.cadc.tap.parser.navigator.SelectNavigator;
+import net.sf.jsqlparser.expression.Expression;
+import net.sf.jsqlparser.expression.Parenthesis;
+import net.sf.jsqlparser.expression.operators.conditional.AndExpression;
+import net.sf.jsqlparser.expression.operators.relational.IsNullExpression;
+import net.sf.jsqlparser.schema.Column;
+import net.sf.jsqlparser.schema.Table;
+import net.sf.jsqlparser.statement.select.PlainSelect;
+import org.apache.log4j.Logger;
+
+/**
+ * Injects a `inventory.storagesite IS NOT NULL` into a WHERE clause.
+ */
+public class SiteLocationsConverter extends SelectNavigator {
+    private static final Logger log = Logger.getLogger(SiteLocationsConverter.class);
+
+    public SiteLocationsConverter() {
+        super(new ExpressionNavigator(), new ReferenceNavigator(), new FromItemNavigator());
+    }
+
+    @Override
+    public void visit(PlainSelect ps) {
+        log.debug("visit(PlainSelect) " + ps);
+        Expression isNotNullExpression = isNotNullExpression();
+        Expression where = ps.getWhere();
+        if (where == null) {
+            ps.setWhere(isNotNullExpression);
+        } else {
+            Parenthesis parenthesis = new Parenthesis(where);
+            Expression and = new AndExpression(parenthesis, isNotNullExpression);
+            ps.setWhere(and);
+        }
+    }
+
+    private Expression isNotNullExpression() {
+        Table artifact = new Table("inventory", "artifact");
+        Column sitelocations = new Column(artifact, "sitelocations");
```

Comment:
This is not the correct column to constrain. Look at the model and documentation to see which things are used at storage sites vs global inventory.

Reply:
Fixed column name.